librz/socket/run.c

```diff
@@ -655,52 +656,130 @@ RZ_API bool rz_run_parseline(RzRunProfile *p, const char *b) {
+			"setenv=FOO=BAR\n"
+			"# unsetenv=FOO\n"
+			"# clearenv=true\n"
+			"# envfile=environ.txt\n"
+			"timeout=3\n"
+			"# timeoutsig=SIGTERM # or 15\n"
+			"# connect=localhost:8080\n"
+			"# listen=8080\n"
+			"# pty=false\n"
+			"# fork=true\n"
+			"# bits=32\n"
+			"# pid=0\n"
+			"# pidfile=/tmp/foo.pid\n"
+			"# #sleep=0\n"
+			"# #maxfd=0\n"
+			"# #execve=false\n"
+			"# #maxproc=0\n"
+			"# #maxstack=0\n"
+			"# #core=false\n"
+			"# #stdio=blah.txt\n"
+			"# #stderr=foo.txt\n"
+			"# stdout=foo.txt\n"
+			"# stdin=input.txt # or !program to redirect input from another program\n"
+			"# input=input.txt\n"
+			"# chdir=/\n"
+			"# chroot=/mnt/chroot\n"
+			"# libpath=$PWD:/tmp/lib\n"
+			"# rzpreload=yes\n"
+			"# preload=/lib/libfoo.so\n"
+			"# setuid=2000\n"
+			"# seteuid=2000\n"
+			"# setgid=2001\n"
+			"# setegid=2001\n"
+			"# nice=5\n"
+			""		
+		);
+	}
+	if (v == 2) {
+		printf(Color_CYAN "Supported RzRun profile options:\n"
+			Color_RESET
+			"arg[0-N]    Set value for argument N passed to the program\n"
+			"aslr        Enable or disable ASLR\n"
+			"bits        Set 32 or 64 bit (if the architecture supports it)\n"
+			"chdir       Change directory before executing the program\n"
+			"chroot      Run the program in chroot. requires some previous setup\n"
+			"clearenv    Unset the whole environment\n"
+			"core        Set no limit the core file size\n"
+			"connect     Connect stdin/stdout/stderr to a socket\n"
+			"pty         Use a pty for connection over socket (with connect/listen)\n"
+			"envfile     Set a file with lines like `var=value` to be used as env\n"
+			"fork        Used with the listen option, allow to spawn a different process for each connection. Ignored when debugging.\n"
+			"input       Set string to be passed to the program via stdin\n"
+			"libpath     Override path where the dynamic loader will look for shared libraries\n"
+			"listen      Bound stdin/stdout/stderr to a listening socket\n"
+			"maxstack    Set the maximum size for the stack\n"
+			"maxproc     Set the maximum number of processes\n"
+			"maxfd       Set the maximum number of file descriptors\n"
+			"nice        Set the niceness level of the process\n"
+			"preload     Preload a library (not supported on Windows, only linux,osx,bsd)\n"
+			"daemon      Set to false by default, otherwise it will run the program in background, detached from the terminal\n"
+			"program     Path to program to be executed\n"
+			"pid         Set to true to print the PID of the process to stderr\n"
+			"pidfile     Print the PID of the process to the specified file\n"
+			"execve      Use execve instead of posix_spawn (osx tricks)\n"
+			"runlib      Path to the library to be executed\n"
+			"runlib.fcn  Function name to call from runlib library\n"
+			"rzpreload   Preload with librz, kill -USR1 to get an rizin shell or -USRZ to spawn a webserver in a thread\n"
+			"rzpreweb    Run the webserver in a thread just at starting the rzpreload\n"
```

Comment:
we should probably rename this and i am not sure if `rzpreweb` is still relevant for us.

Reply:
Yes, should be removed, that feature was removed long time ago. Also from the `man` page.